### PR TITLE
feat: add delete users_events_roles endpoint

### LINKED ISSUE #7558: Add functionality to delete User Event Role
**Is your feature request related to a problem? Please describe.**
Currently there is no functionality to delete User Event Role. Adding this functionality can allow event organizers to delete role assigned to any user.

Ref issue: #7235

### PR DESCRIPTION
**Pull request by Tlazypanda**
Fixes #7558 

#### Short description of what this resolves:
Added delete users_events_roles endpoint as per the resource manager architecture.

#### Changes proposed in this pull request:

- Added the schema
- Added the api
- Added the route

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Should also have a list API. /events/I'd/user-event-roles

**Comment by Tlazypanda** [comment]
Should that also have the same set of permissions as this one? @iamareebjamal  also I have to make add factory tests and make a new module for User Event Role in the Api frontend docs

**Comment by iamareebjamal** [comment]
No, user event roles cannot be created, only queried, patched and deleted. Factory and Dredd tests would be good but are optional

**Comment by Tlazypanda** [comment]
@iamareebjamal Actually I meant permissions in the sense only admin and owner should have access to these endpoints right? :sweat_smile:

**Comment by iamareebjamal** [comment]
co organizer

**Comment by Tlazypanda** [comment]
Alright thanks :v: on it!

**Comment by iamareebjamal** [comment]
@Tlazypanda Please don't force push unless rebasing, because it makes it difficult to know incremental changes

**Comment by iamareebjamal** [comment]
Great work BTW

**Comment by iamareebjamal** [comment]
The getting list of UER should be /events/ID/user-event-roles/

Getting all roles like /user-event-roles/ shouldn't be possible

**Comment by Tlazypanda** [comment]
@iamareebjamal I have added the route in events/ID/user-event-roles/ form only and the /users-events-roles route is not present. Am I missing it somewhere? :sweat_smile: Also I am not sure how to pass this dredd test it seems to be working locally

**Comment by iamareebjamal** [comment]
See other patch hooks, you have to create a UER first

File: tests/hook_main.py [hunk 1]
```diff
@@ -3447,7 +3447,19 @@ def users_events_roles_patch(transaction):
     :return:
     """
     with stash['app'].app_context():
-        users_events_roles = UsersEventsRolesFactory()
+        role = RoleFactory()
+        db.session.add(role)
+        db.session.commit()
```
Comment (iamareebjamal) [inline tests/hook_main.py:3452]: Commit once. And I think these are automatically added to session

File: tests/hook_main.py [hunk 2]
```diff
@@ -3447,7 +3447,19 @@ def users_events_roles_patch(transaction):
...
+
+        user = UserFactory()
+        db.session.add(user)
+        db.session.commit()
+
+        event = EventFactoryBasic()
+        db.session.add(event)
+        db.session.commit()
+
+        users_events_roles = UsersEventsRolesFactory(event_id=1, user_id=1, role_id=1)
```
Comment (iamareebjamal) [inline tests/hook_main.py:3462]: Instead, create a sub factory. This will create all of the above automatically

**Comment by Tlazypanda** [comment]
@iamareebjamal tests are still failing is this the way I was supposed to write the subfactory test right? :sweat_smile:

**Comment by Tlazypanda** [comment]
@iamareebjamal tests passing now yayy!! :v:

**Comment by iamareebjamal** [comment]
Great Work! Next step is to verify that /events/ID?include=user-event-roles.role works, but that can be another PR

**Comment by Tlazypanda** [comment]
@iamareebjamal  Ohh sure so adding integration test for that should be the issue then right? :sweat_smile:

**Comment by iamareebjamal** [comment]
Just verify if it is working, if it is not, then fix it. No need for tests

**Comment by Tlazypanda** [comment]
@iamareebjamal the url is not working since users_events_role is not added as a relationship for event should I open an issue for adding it? :sweat_smile:

**Comment by iamareebjamal** [comment]
Yes

**Pull request merged**